docs/rfc/rfc-8.adoc

```diff
@@ -0,0 +1,410 @@
+
+Each native bridge has its own way mechanism to manage an Ethereum-to-chain
+token contract mapping, sometimes called a token registry.
+
+Examples of chains with native bridges include:
+
+* Arbitrum,
+* Polygon,
+* Optimism.
+
+==== Arbitrum native bridge
+
+Since our tBTC L2 token is upgradeable and will allow multiple minting
+authorities, the Arbitrum Standard ERC20 Gateway is not sufficient.
+
+Given our tBTC L1 token does not conform to `ICustomToken` interface, we can not
+use Arbitrum Generic Custom Gateway.
+
+The only option is to implement our custom gateway that will have a minting
+authority for Arbitrum L2 tBTC token.
+
+One working example of a custom gateway is the DAI gateway:
+
+* The L1 gateway contract: link:https://etherscan.io/address/0xD3B5b60020504bc3489D6949d545893982BA3011[`L1DaiGateway`],
+* The L2 gateway contract: link:https://arbiscan.io/address/0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65[`L2DaiGateway`].
+
+Another working example of a custom gateway is the Livepeer gateway:
+
+* The L1 gateway contract: link:https://etherscan.io/address/0x6142f1C8bBF02E6A6bd074E8d564c9A5420a0676[`L1LPTGateway`],
+* The L2 gateway contract: link:https://arbiscan.io/address/0x6D2457a4ad276000A615295f7A80F79E48CcD318[`L2LPTGateway`].
+
+This is out of the scope of the initial implementation.
+
+See link:https://developer.arbitrum.io/asset-bridging[Arbitrum documentation for token bridging]
+for more information.
+
+==== Polygon native bridge
+
+For similar reasons as for Arbitrum, we have to implement our custom `FxTunnel`
+implementation. The child tunnel implementation on the Polygon sidechain will
+have a minting authority for the Polygon tBTC token.
+
+This is out of the scope of the initial implementation.
+
+See link:https://wiki.polygon.technology/docs/develop/l1-l2-communication/fx-portal[Polygon FxPortal documentation]
+for more information.
+
+==== Optimism native bridge
+
+The Optimism Standard Bridge creates L2 token with a call to
+`OptimismMintableERC20Factory`. The created token does not have the capabilities
+we expect from tBTC canonical tokens on L2. For this reason, using the Optimism
+Standard Bridge is not an option and we must implement our own L2 minter
+contract with an authority to mint Optimism tBTC. Note that going back from L2
+to L1 will take at least one week given the one-week Optimism challenge period.
+
+This is out of the scope of the initial implementation.
+
+See link:https://github.com/ethereum-optimism/optimism-tutorial/tree/main/standard-bridge-standard-token[Optimism guide for the standard bridge]
+and link:https://community.optimism.io/docs/developers/bridge/messaging/#fees-for-l2-%E2%87%92-l1-transactions[Optimism documentation on sending data between L1 and L2]
+for more details.
+
+=== Supporting cross-ecosystem bridges
+
+The Wormhole bridge will be the first cross-ecosystem bridge to have a minting
+authority for tBTC token on Arbitrum, Polygon, and Optimism.
```

Comment:
What about other EVM chains and non-EVM chains?  Without a canonical contract on other chains, the Wormhole token representation (or another bridge) will be used and may stay that way for a long time if a lot of liquidity sinks in.  Is there a plan to try to migrate this or recommend a bridge as canonical?

Reply:
For non-EVM chains, I think we need a canonical bridge, period. We won't have the bandwidth to maintain those chains for the forseeable future.

 For other EVMs, though, an ideal solution would either be...
 * To stage the rollout so we can deploy canonical contracts to each one
 * To work with a canonical bridge that agrees to cede control of the token contracts when the team is ready

Reply:
I don't think it would be feasible for bridges (at least wormhole) to cede control of one of their assets.  It would mess with a number of security assumptions.

Reply:
> I don't think it would be feasible for bridges (at least wormhole) to cede control of one of their assets. It would mess with a number of security assumptions.

This is a day 1 feature at Nomad

https://github.com/nomad-xyz/monorepo/blob/main/packages/contracts-bridge/contracts/TokenRegistry.sol#L141-L166

Reply:
rather than ceding control, it enrolls a team-controlled token as the bridge token, and allows users to migrate to the new token